mavros/src/lib/uas_stringify.cpp

```diff
@@ -316,3 +316,39 @@ std::string UAS::str_system_status(enum MAV_STATE st)
 	return state_strings[idx];
 }
 
+static const std::array<const std::string, 12> frame_strings = {
+	/*  0 */ "GLOBAL",
+	/*  1 */ "LOCAL_NED",
+	/*  2 */ "MISSION",
+	/*  3 */ "GLOBAL_RELATIVE_ALT",
+	/*  4 */ "LOCAL_ENU",
+	/*  5 */ "GLOBAL_INT",
+	/*  6 */ "GLOBAL_RELATIVE_ALT_INT",
+	/*  7 */ "LOCAL_OFFSET_NED",
+	/*  8 */ "BODY_NED",
+	/*  9 */ "BODY_OFFSET_NED",
+	/* 10 */ "GLOBAL_TERRAIN_ALT",
+	/* 11 */ "GLOBAL_TERRAIN_ALT_INT"
+};
+
+std::string UAS::str_frame(enum MAV_FRAME fr)
+{
+	size_t idx = size_t(fr);
+	if (idx >= frame_strings.size())
+		return std::to_string(idx);
+
+	return frame_strings[idx];
+}
+
+uint8_t UAS::idx_frame(std::string mav_frame)
+{
+	uint8_t index, idx = 0;
+	for (const auto &frame : frame_strings) {
+		if (frame.compare(mav_frame) == 0)
```

Comment:
Well this was implemented some time ago, and I didn't check it this time. I got used to compare somewhat :)